types/wlr_surface.c

```diff
@@ -55,6 +82,7 @@ static void surface_commit(struct wl_client *client,
 			} else {
 				uint32_t format = wl_shm_buffer_get_format(buffer);
 				wlr_texture_upload_shm(surface->texture, format, buffer);
+				wl_resource_queue_event(surface->pending_buffer, WL_BUFFER_RELEASE);
```

Comment:
looked the functions up and it boils down to whether to write the release message immediately or write it later (like in "queue"). So i guess both work, nvm go with the weston approach then.